src/core/login/pin2.ts

```diff
@@ -173,14 +173,18 @@ export async function changePin(
 export async function checkPin2(
   ai: ApiInput,
   login: LoginTree,
-  pin: string
+  pin: string,
+  forDuressAccount?: boolean
 ): Promise<boolean> {
-  const { appId, loginId, username } = login
+  const { loginId, username } = login
   if (username == null) return false
 
+  const appId =
+    forDuressAccount === true ? login.appId + '.duress' : login.appId
+
   // Find the stash to use:
   const { stashTree } = getStashById(ai, loginId)
-  const stash = findPin2Stash(stashTree, appId)
+  const stash = findPin2StashSpecific(stashTree, appId)
```

Comment:
It seems like this would break the existing logic.

I would have expected:

```ts
const stash = forDuressAccount
  ? searchTree(stashTree, stash => stash.appId === appId + '.duress')
  : findPin2Stash(stashTree, appId)
```

Then the existing code-path stays unchanged.

Of course, we'd also want to fix the `.duress.duress` logic error mentioned earlier.

Reply:
This would fail because if the loginStash is the `.duress` account, then `searchTree` would fail to find it in that case. `findPin2StashSpecific` is invented for this reason. "Specific" here is meant to mean target "appId" specifically passed; `findPin2Stash` doesn't return the stash specifically for appId.

Reply:
```
const stash = forDuressAccount ? findPin2StashSpecific(stashTree, appId) : findPin2Stash(stashTree, appId)
```